cdap-docs/_common/_themes/cdap/static/version-menu.js

```diff
@@ -35,43 +35,58 @@
   var buildURL = (function(dir){
     return versionsURL + dir + '/en/';
   });
-  var writelink = (function(dir, label){
-    document.write('<option value="' + buildURL(dir) + '">Version ' + label + '</option>');
+  var writeLink = (function(dir, label){
+    document.write('<option value="' + buildURL(dir) + '">' + label + '</option>');
+  });
+  var writeVersionLink = (function(dir, label){
+    writeLink(dir, 'Version ' + label);
   });
   window.versionscallback = (function(data){
+    var ess;
     if (data) {
       document.write('<li class="versions">');
       document.write('<select id="' + versionID + '" onmousedown="window.currentversion=this.value;" onchange="window.gotoVersion(\'' + versionID + '\')">');
     }
-    var ess = "s";
-    if (data.development && data.development.length > 0) {
-      if (data.development.length == 1) {
-        ess = "";
-      }
+    
+    if (data.development && data.development.length > 0 && data.development[0]) {
+      ess = (data.development.length == 1) ? "" : "s" ;
```

Comment:
If `data.development.length` is `0`, then the entire block is skipped per line 51 and instead a link to develop is inserted.

As for the structure of the "if", it only does the singular for a single instance; if there were zero, it (was) still plural...though that has been removed as an option now. I guess I just think of that way: the ess is added if it is not  singular.